docs/sources/setup-grafana/configure-security/configure-authentication/saml.md

```diff
@@ -407,3 +407,20 @@ The keys you provide should look like:
 ...
 -----END PRIVATE KEY-----
 ```
+
+### SAML login attempts fail with request response "origin not allowed"
+
+When the user logs in using SAML and gets presented with "origin not allowed", the user might be issuing the login from an IdP (identity provider) service or the user is behind a reverse proxy. This potentially happens as Grafana's CSRF checks deem the requests to be invalid. For more information [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery).
+
+To solve this issue, you can configure either the [`csrf_trusted_origins`]({{< relref "../../configure-grafana/enterprise-configuration/#csrf-trusted-origins" >}}) or [`csrf_origin_headers`]({{< relref "../../configure-grafana/enterprise-configuration/#csrf-origin-headers" >}}) option in the SAML configuration.
+
+Example of a configuration file:
+
+```bash
+# config.ini
+...
+[auth.saml]
+csrf_trusted_origins = https://grafana.example.com
+csrf_origin_headers = X-Forwarded-Proto X-Forwarded-Host X-Forwarded-Port
```

Comment:
Is `csrf_origin_headers` the correct option to use here? `csrf.go` uses only `csrf_additional_headers` and `csrf_trusted_origins`.

https://github.com/grafana/grafana/blob/d32ae1890971343fe24ea9bf098bca049992c3dd/pkg/middleware/csrf/csrf.go#L38-L39

Reply:
it is not! thanks for pointing it out!

Reply:
https://github.com/grafana/grafana/pull/50589